arxivdigest/api/database.py

```diff
@@ -295,27 +294,42 @@ def get_user_feedback_articles(ids):
 
 def get_user_feedback_topics(ids):
     """Returns topic feedback data for the requested userIDs in this format:
-    {userid:
-        date: [
-            {topic: {
-                'seen': date or None,
-                'clicked': date or None,
-                }, ...
+    {userid: {
+        topic: {
+            'seen': date, None or not existing,
+            'clicked': date, None or not existing,
+            'state': varchar, None or not existing,
+            'recommendation_date': date, None or not existing,
+            'interaction_date': date, None or not existing
             }, ...
-        ], ...
+        }, ...
     }
     """
     with closing(getDb().cursor(dictionary=True)) as cur:
-        sql = """SELECT user_id, topic, DATE(datestamp) as date, seen, clicked
-                 FROM topic_recommendations tr NATURAL JOIN topics t
-                 WHERE user_id IN ({}) AND interleaving_order IS NOT 
-                 null ORDER BY  interleaving_order 
-                 DESC""".format(','.join(['%s'] * len(ids)))
+        tr_sql = '''SELECT tr.user_id, t.topic, tr.datestamp 
```

Comment:
should include interleaving order in result, as its generally not okay to trust the sort order of a dict.